src/02-room-details.js

```diff
@@ -49,7 +73,31 @@ function getRoomByDinosaurName(dinosaurs, rooms, dinosaurName) {}
       "Kit Hopkins Education Wing"
     ]
  */
-function getConnectedRoomNamesById(rooms, id) {}
+function getConnectedRoomNamesById(rooms, id) {
+  let results = [] //Holds the answer
+  let correctId = "" //Holds id 
+  let incorrectIdFound = false//Holds false as default
+  for (let i = 0; i < rooms.length; i++){ //Loops through rooms array
+    if(rooms[i].roomId === id){ //Checks if id is found in rooms
+      correctId = id //Used to see if id is found in rooms
+      for (let j = 0; j < rooms[i].connectsTo.length; j++){ //Loops through connectsTo array
+        for (let k = 0; k < rooms.length; k++){ //Loops through rooms array again 
+          if (rooms[i].connectsTo[j] === rooms[k].roomId){ //Checks if any connectsTo id in rooms is matching any id in rooms     
+            results.push(rooms[k].name) //Pushes the name of the room into the answer
+          } else if (rooms[i].connectsTo[j] === 'incorrect-id'){ //Checks if any connectsTo id is labled as "incorrect-id"
+            incorrectIdFound = true //Updates to true so we know it is found
+          }
+        }
+      }
+    }
+  }
+  if (correctId !== id){ // Checsk if correctId was updated
+    return `Room with ID of '${id}' could not be found.` //Returns error message if the id was not found in the array of rooms
+  } else if (incorrectIdFound === true){ //Checks if incorrectIdFound was updated
+    return `Room with ID of 'incorrect-id' could not be found.` //Returns error message if the id in connectsTo is incorrect
```

Comment:
this is considered hardcoded. What if the id was `abc1234`, we would still want to get this error message with the incorrect id dynamically inserted in the string. 

In order to check which ID is incorrect you would have to look for each ID in the `rooms` array and the one that is not found you would return the error message